cpp/src/arrow/filesystem/s3fs_narrative_test.cc

```diff
@@ -68,8 +68,13 @@ std::shared_ptr<FileSystem> MakeFileSystem() {
   options.endpoint_override = FLAGS_endpoint;
   options.scheme = FLAGS_scheme;
   options.region = FLAGS_region;
-  s3fs = S3FileSystem::Make(options).ValueOrDie();
-  return std::make_shared<SubTreeFileSystem>(FLAGS_bucket, s3fs);
+  options.allow_bucket_creation = true;
```

Comment:
```suggestion
  options.allow_bucket_creation = FLAGS_create;
```

Reply:
Shouldn't this always be true for these tests?